File: web/src/api/platform-audit.ts
import { http } from '../lib/http'

export interface PlatformAuditEvent {
  id: number
  enterprise_space_id: number | null
  user_id: number | null
  username: string | null
  space_name: string | null
  action: string
  resource_type: string
  resource_id: string | null
  request_id: string | null
  ip_address: string | null
  detail: Record<string, unknown> | null
  message: string | null
  created_at: string
}

export interface PlatformAuditListResponse {
  items: PlatformAuditEvent[]
  total: number
  skip: number
  limit: number
}

export interface PlatformAuditListParams {
  enterprise_space_id?: number
  user_id?: number
  action?: string
  resource_type?: string
  created_from?: string
  created_to?: string
  skip?: number
  limit?: number
}

export async function listPlatformAuditEvents(params?: PlatformAuditListParams) {
  return http.get<PlatformAuditListResponse>('/platform-audit/events', { params })
}

export const PLATFORM_AUDIT_ACTION_LABELS: Record<string, string> = {
  'auth.login': '用户登录',
  'admin.user.create': '创建用户',
  'admin.user.update': '更新用户',
  'admin.user.delete': '删除用户',
  'admin.user.memberships.update': '更新用户空间分配',
  'knowledge_base.create': '创建知识库',
  'knowledge_base.update': '更新知识库',
  'knowledge_base.delete': '删除知识库',
  'model.provider.create': '创建模型 Provider',
  'model.provider.update': '更新模型 Provider',
  'model.provider.delete': '删除模型 Provider',
  'model.enabled.update': '更新启用模型',
  'model.defaults.update': '更新默认模型',
  'chat.session.delete': '删除对话会话',
  'chat.conversation.delete': '删除对话记录',
  'audit.test': '审计测试',
}

export const PLATFORM_AUDIT_RESOURCE_TYPE_LABELS: Record<string, string> = {
  user: '用户',
  knowledge_base: '知识库',
  model: '模型',
  chat: '对话',
  system: '系统',
}

export function formatAuditAction(action: string) {
  return PLATFORM_AUDIT_ACTION_LABELS[action] || action
}

export function formatAuditResourceType(resourceType: string) {
  return PLATFORM_AUDIT_RESOURCE_TYPE_LABELS[resourceType] || resourceType
}

File: backend/app/api/routes/platform_audit.py
from datetime import datetime

from fastapi import APIRouter, Depends, HTTPException, Query, Request, status
from sqlalchemy.orm import Session

from app.core.enterprise_space_context import CurrentSpace, CurrentUser, get_membership
from app.core.membership_roles import SPACE_ADMIN
from app.db.session import get_db
from app.models.enterprise_space import EnterpriseSpace, Membership, User
from app.schemas.platform_audit import PlatformAuditEventListResponse, PlatformAuditEventResponse
from app.services.platform_audit_service import list_platform_audit_events, record_platform_audit

router = APIRouter(prefix="/platform-audit", tags=["platform-audit"])


def _require_space_admin_or_system(
    current_user: User,
    current_space: EnterpriseSpace,
    db: Session,
) -> Membership | None:
    """系统管理员直接放行；否则要求当前空间 space_admin。"""
    if current_user.is_admin:
        return get_membership(current_user, current_space, db)
    membership = get_membership(current_user, current_space, db)
    if membership is None or membership.role != SPACE_ADMIN:
        raise HTTPException(
            status_code=status.HTTP_403_FORBIDDEN,
            detail="只有企业空间管理员或系统管理员可以执行此操作",
        )
    return membership


@router.get("/events", response_model=PlatformAuditEventListResponse)
def list_audit_events(
    current_user: CurrentUser,
    current_space: CurrentSpace,
    db: Session = Depends(get_db),
    enterprise_space_id: int | None = Query(default=None, description="平台管理员可选，按企业空间筛选"),
    user_id: int | None = Query(default=None, description="按操作用户筛选"),
    action: str | None = Query(default=None),
    resource_type: str | None = Query(default=None),
    created_from: datetime | None = Query(default=None),
    created_to: datetime | None = Query(default=None),
    skip: int = Query(default=0, ge=0),
    limit: int = Query(default=20, ge=1, le=200),
) -> PlatformAuditEventListResponse:
    _require_space_admin_or_system(current_user, current_space, db)

    if current_user.is_admin:
        scoped_space_id = enterprise_space_id
        if scoped_space_id is not None and db.get(EnterpriseSpace, scoped_space_id) is None:
            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="企业空间不存在")
    else:
        scoped_space_id = current_space.id

    items, total = list_platform_audit_events(
        db,
        enterprise_space_id=scoped_space_id,
        user_id=user_id,
        action=action,
        resource_type=resource_type,
        created_from=created_from,
        created_to=created_to,
        skip=skip,
        limit=limit,
    )
    return PlatformAuditEventListResponse(items=items, total=total, skip=skip, limit=limit)


@router.post("/events/test", response_model=PlatformAuditEventResponse, include_in_schema=False)
def create_test_audit_event(
    request: Request,
    current_space: CurrentSpace,
    current_user: CurrentUser,
    db: Session = Depends(get_db),
) -> PlatformAuditEventResponse:
    _require_space_admin_or_system(current_user, current_space, db)
    row = record_platform_audit(
        db,
        action="audit.test",
        resource_type="system",
        enterprise_space_id=current_space.id,
        user_id=current_user.id,
        request_id=request.headers.get("X-Request-ID"),
        ip_address=request.client.host if request.client else None,
        message="audit pipeline smoke test",
    )
    db.commit()
    db.refresh(row)
    return PlatformAuditEventResponse.model_validate(row)
